src/odemis/driver/technolution.py

```diff
@@ -823,16 +823,32 @@ def _setResolution(self, resolution):
         number of cell images (detector shape) multiplied with the size of a cell image (effective cell size). Method
         sets the resolution of a single field image, so that the effective cell size is a multiple of 4 (in both x
         and y direction).
+        The following requirements need to be met: effective cell size <= cell complete size - max cell translation
 
         :param resolution (int, int): requested resolution for a single field image. It is the product of the number of
         cell images multiplied by the effective cell size.
         :return (int, int): resolution closest possible to the requested resolution for the single field image.
         """
         COMMON_DIVISOR = 4
 
-        req_eff_cell_size = numpy.array(resolution) / self.parent._mppc.shape[0:2]  # Requested effective cell size
+        eff_cell_size = numpy.array(resolution) / self.parent._mppc.shape[0:2]  # effective cell size
         # Round to closest multiple of COMMON_DIVISOR (x.5 is rounded down)
-        eff_cell_size = numpy.round(req_eff_cell_size / COMMON_DIVISOR) * COMMON_DIVISOR
+        eff_cell_size = numpy.round(eff_cell_size / COMMON_DIVISOR) * COMMON_DIVISOR
+
+        # check: effective cell size <= cell complete size - max cell translation = minimal effective cell size;
+        max_cell_translation = max(max(self.parent._mppc.cellTranslation.value))
+        min_eff_cell_size = (self.parent._mppc.cellCompleteResolution.value[0] - max_cell_translation[0],
+                             self.parent._mppc.cellCompleteResolution.value[1] - max_cell_translation[1])
+
+        if tuple(eff_cell_size) > min_eff_cell_size:
+            raise ValueError("(Requested resolution of %s is too big. The resolution is the effective cell size "
+                             "multiplied by the detector shape. The effective cell size must be equal or smaller than "
+                             "the complete cell size minus the maximum cell translation value. "
+                             "The minimal resolution possible with the currently settings for cell complete resolution "
+                             "and cell translation is %s.\n "
+                             "Resolution values remain unchanged."
```

Comment:
That's a really long error message. Can you make it more compact? At least, the last not about unchanged value can go away: it's the expected behaviour if a "=" fails, and it's what happens with all the VAs.
Maybe replace the english by just mathematical formula/pseudo code?